packages/dds/map/src/mapKernel.ts

```diff
@@ -548,6 +548,18 @@ export class MapKernel {
         return previousLocalValue;
     }
 
+    private getClearLocalOpMetadata(op: IMapClearOperation, previousMap?: Map<string, ILocalValue>): unknown {
+        return { type: "clear", pendingMessageId: this.getMapClearMessageId(), previousMap };
+    }
+
+    private getKeyLocalOpMetadata(op: IMapKeyOperation, previousValue?: ILocalValue): unknown {
+        const pendingMessageId = this.getMapKeyMessageId(op);
+        const localMetadata = previousValue ?
+            { type: "edit", pendingMessageId, previousValue } :
```

Comment:
With this change, please update the type of IMapKeyEditLocalOpMetadata.previousValue so that undefined is not allowed. (It used to be allowed for the stashed op case because we weren't filling out the local op metadata fully before.)